launcher/src/backend/NodeConnection.js

```diff
@@ -777,10 +777,27 @@ export class NodeConnection {
     })
   }
 
-  async destroyNode() {
+  async destroyNode(serviceConfigs) {
+
     const ref = StringUtils.createRandomString();
     this.taskManager.tasks.push({ name: "Delete Node", otherRunRef: ref });
 
+    let sPathCheck = "";
+    for(let service of serviceConfigs) {
+      let sID = service.id;
+      for(let path of service.volumes) {
+        let sPath = path.destinationPath.substring(0, path.destinationPath.indexOf(sID)) + sID + "/";
+        if(sPath.includes(sID) && !sPath.includes("/etc/stereum") && sPath != sPathCheck){
+          sPathCheck = sPath;
+          this.taskManager.otherSubTasks.push({
+            name: "remove Volume Directories",
+            otherRunRef: ref,
+            status: !(await this.sshService.exec("rm -rf " + sPath)).rc,
```

Comment:
this would make a subtask for every volume which is fine if you give them different names. in this case they're all named `remove Volume Directories`